packages/jest/src/resultsProcessor.ts

```diff
@@ -0,0 +1,126 @@
+
+// Map of test suite name to test results
+const consolidatedErrors = new Map<string, AssertionResult[]>();
+
+/**
+ * Create a test failure result for given a11y failure
+ */
+function convertA11yTestResult(testResult: AssertionResult, a11yResult: A11yResult): AssertionResult {
+    return {
+        ...testResult,
+        // TODO (refactor): extract formatting into its own function.
+        //  - Can this satisfy Formatter interface?
+        //  - Be part of format? (FileFormatter vs ConsoleFormatter)?
+        fullName: `${a11yResult.description}: ${a11yResult.selectors}`,
+        failureMessages: [
+            `${errMsgHeader}: ${a11yResult.description}
+CSS Selectors: ${a11yResult.selectors}
+HTML element: ${a11yResult.html}
+Help: ${a11yResult.helpUrl}
+Tests: "${testResult.fullName}"
+Summary: ${a11yResult.summary}`, // TODO (refactor): replace with array of tests when consolidating? But we reach here after de-duping a11y results?
+        ],
+        // We don't need the error objects anymore as they have been processed
+        failureDetails: [],
+        // Add all test's having the same a11y issue
+        ancestorTitles: [...new Set(testResult.ancestorTitles).add(testResult.fullName)],
+    } as AssertionResult;
+}
+
+/**
+ * Convert any a11y errors from test failures into their own test suite, results
+ */
+function processA11yErrors(testSuite: TestResult, testResult: AssertionResult) {
+    const suiteName = testSuite.testFilePath.substring(testSuite.testFilePath.lastIndexOf('/') + 1);
+
+    testResult.failureDetails.forEach((failure) => {
+        let error = (failure as FailureDetail).error;
+        // If using circus test runner https://github.com/facebook/jest/issues/11405#issuecomment-843549606
+        // TODO (code cov): Add test data covering the case for circus test runner
+        /* istanbul ignore next */
+        if (error === undefined) error = failure as A11yError;
+        if (error.name === A11yError.name) {
+            // TODO : What happens if there are ever multiple failureDetails?
+            //  Ideally there shouldn't be as test execution should be stopped on failure
+            ConsolidatedResults.consolidate(error.a11yResults, suiteName).forEach((a11yResult) => {
+                const suiteKey = `[Sa11y ${a11yResult.wcag} ${a11yResult.id} ${suiteName}]`;
+                // TODO (code cov): Fix - should be covered by existing tests
+                /* istanbul ignore next */
+                if (!Array.isArray(consolidatedErrors.get(suiteKey))) consolidatedErrors.set(suiteKey, []);
+                consolidatedErrors.get(suiteKey)?.push(convertA11yTestResult(testResult, a11yResult));
+            });
+        }
+    });
+}
+
+/**
+ * Modify existing test suites, test results containing a11y errors after a11y errors
+ *  are extracted out into their own test suites and results.
+ */
+function modifyTestSuiteResults(results: AggregatedResult, testSuite: TestResult, testResult: AssertionResult) {
```

Comment:
I don't really understand what the purpose of this function is. Can you explain more?

Reply:
Since the custom reporter is transforming the a11y errors into their own test failures contained within a new suite based on the a11y rule (const suiteKey = `[Sa11y ${a11yResult.wcag} ${a11yResult.id} ${suiteName}]`;), the original test in which the a11y error  occurred is being disabled and test failure counts adjusted accordingly. Guess if the main logic is refactored to use map/filter without using any intermediates, it might be possible to just filter these out instead of disabling (the test counts will still need to be adjusted I think).

I have updated the PR desc to include docs/screenshots to illustrate this. Please check them out and let me know if it makes sense and if we can include them as part of jest README.

Reply:
Ok. I just want to make sure we don't mess up the overall test and failure counts. E.g., If we have a functional failure and then that is fixed but replaced by an a11y failure, then the total tests and failure counts should remain the same. 

It sounds like you're accounting for that. Just wanted to make sure.

Reply:
That is a good point. The failure count and the total test count could increase with the sa11y results processor - because each a11y failure from a test would be extracted to be its own test failure. When there is only 1 a11y failure in a test, the count would stay the same. But when there are multiple unique a11y failures in a test, the count could increase. But the results processor doesn't affect default console reporter or other reporters - it affects just the JSON output. Added a limitations section to the doc to reflect this and other points - can you please review.

Also created https://github.com/salesforce/sa11y/issues/66 as it is important and deserves its own issue.